algebra/direct_limit.lean

```diff
@@ -0,0 +1,275 @@
+import algebra.ring order.lattice
+
+universes u v
+
+theorem congr_arg₂ {α β γ : Type*} (f : α → β → γ) {x₁ x₂ : α} {y₁ y₂ : β}
+  (Hx : x₁ = x₂) (Hy : y₁ = y₂) : f x₁ y₁ = f x₂ y₂ :=
+eq.drec (eq.drec rfl Hy) Hx
```

Comment:
This is in the wrong place, but I don't know why you need it. `by congr` should do the job